### PR TITLE
isBotInstalledUsingRoomId

### PR DESCRIPTION
**Pull request by diogo-pessoa**
Description
---

applying changes to use roomID, discussed at [PR](https://github.com/SFDC/refocus-bot-ttx/pull/40)

### CODE REVIEW

File: tests/getRoomById.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+  it('OK should fail when invalid id', async () => {
+    getGenericRoomStub.returns(Promise.resolve(room));
+    const roomBody = await bdk.getRoomById();
+    // eslint-disable-next-line no-unused-expressions
+    expect(roomBody).to.be.null;
+  });
+
+  it('OK should be null in when request fails', async () => {
+    getGenericRoomStub.returns(Promise.resolve());
```
Comment (cassiodias) [inline tests/getRoomById.js:55]: is this making the flow going to the catch?

File: tests/isBotInstalled.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/* eslint-disable no-unused-expressions */
+/* eslint-disable no-unused-expressions,max-len */
```
Comment (cassiodias) [inline tests/isBotInstalled.js:1]: I think lint could be disabled inside tests

File: refocus-bdk-server.js [hunk 3]
```diff
@@ -486,42 +486,41 @@ module.exports = (config) => {
...
    * @returns {boolean}
    */
-  async function isBotInstalledInRoom(event, bot) {
-    const eventRoomTypeId = event.Room.RoomType.id;
+  async function isBotInstalledInRoom(roomId, bot) {
     try {
-      const botsInRoomType = await getRoomTypeById(eventRoomTypeId);
-      if (botsInRoomType) {
-        const botsInstalled = botsInRoomType.bots.find((b) => b === bot);
+      const botsInRoom = await getRoomById(roomId);
```
Comment (LongweiDeng) [inline refocus-bdk-server.js:516]: The var name `botsInRoom` doesn't sound right to me, as it assigns the room response body (not the list of bots). Should it be named `room` instead?

File: refocus-bdk-server.js [hunk 4]
```diff
@@ -486,42 +486,41 @@ module.exports = (config) => {
...
+      if (botsInRoom) {
+        const botsInstalled = botsInRoom.bots.find((b) => b === bot);
```
Comment (LongweiDeng) [inline refocus-bdk-server.js:518]: "The find() method returns the value of the first element in the provided array that satisfies the provided testing function."
How about var name `installedBot` or `botInstalled,` as singleton?

Comment (LongweiDeng) [reply refocus-bdk-server.js:518]: Alternatively, you could also do
`const botsInstalled = room.bots`   (assuming we agree to rename the 'botsInRoom' to 'room')
`return botsInstalled.includes(bot)`

File: tests/getRoomById.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  let getGenericRoomStub;
+  beforeEach(() => {
+    getGenericRoomStub = sinon.stub(generic, 'get');
+  });
+
+  afterEach(() => {
+    getGenericRoomStub.restore();
+  });
+
+  it('OK if roomTypeId matches response', async () => {
```
[the remark this answers is not in the document]
Comment (diogo-pessoa) [reply tests/getRoomById.js:40]: should be roomId, I knew I'd forget to change it somewhere.

File: tests/getRoomById.js (hunk 1 above)
Comment (diogo-pessoa) [reply tests/getRoomById.js:55]: yes, this causes an undefined from the get (bad response) and is handled by the catch returning nulll.

File: tests/isBotInstalled.js (hunk 2 above)
Comment (diogo-pessoa) [reply tests/isBotInstalled.js:1]: changed to disable inside the tests.

File: refocus-bdk-server.js (hunk 3 above)
Comment (diogo-pessoa) [reply refocus-bdk-server.js:516]: I'll change the var name it makes more sense, as the response is just the room payload.

File: tests/getRoomById.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    getGenericRoomStub.returns(Promise.resolve(room));
+    const roomId = 2;
+    const roomBody = await bdk.getRoomById(roomId);
```
[the remark this answers is not in the document]
Comment (diogo-pessoa) [reply tests/getRoomById.js:43]: I see what you mean now that you pointed that, thanks, I'll change it

File: refocus-bdk-server.js (hunk 4 above)
Comment (diogo-pessoa) [reply refocus-bdk-server.js:518]: Cheers, I'm going with the second approach you suggested, makes the reading clearer that we want to know if bot x is installed in that room.

**Pull request merged**